### PR TITLE
WP Scripts: Make watch mode more resilient for developer errors

### PR DESCRIPTION
**Pull request by gziolo**
## What?

Extracted from https://github.com/WordPress/gutenberg/pull/66466#discussion_r1816548283:

> @sirreal, I checked the previous implementation before the function had support for `buildType` added in https://github.com/WordPress/gutenberg/pull/57461 and the `return` was inside the callback rather than loop. So it should continue execution and skip only misconfigured entries.

## Why?

This was useful in the watch mode when running `wp-scripts start` as it allowed developers made mistakes during development. There would be an error message displayed in the terminal that should help to resolve the issue without running the same command again.

## Testing Instructions

I created a custom block from the Gutenberg repository and put it in the watch mode with:
```bash
npx wp-create-block example-block --no-wp-scripts
cd example-block
npx ../node_modules/.bin/wp-scripts start
```

I made an error in `block.json` file to ensure it errors. I was able to see the warning coming from webpack config and the error coming from the bundler for the usage of import inside the JS code for `block.json`. The command continued to work and resumed to fully operational after I fixed the JSON file and saved it again.

Some examples after the changes applied with this PR while webpack continues to run in the watch mode:

Screenshot 2024-11-05 at 15 13 04

Screenshot 2024-11-05 at 15 13 11

### CODE REVIEW

**Comment by gziolo** [comment]
For reference, the previous contribution from @fabiankaegy which added improved support for `block.json` files:
- https://github.com/WordPress/gutenberg/pull/51971

File: packages/scripts/utils/config.js [hunk 1]
```diff
@@ -370,35 +363,44 @@ function getPhpFilePaths( context, props ) {
...
 	return blockMetadataFiles.flatMap( ( blockMetadataFile ) => {
-		const blockJson = JSON.parse( readFileSync( blockMetadataFile ) );
-
 		const paths = [];
+		let parsedBlockJson;
+		try {
+			parsedBlockJson = JSON.parse( readFileSync( blockMetadataFile ) );
+		} catch ( error ) {
+			warn(
+				`Skipping scanning "${ blockMetadataFile.replace(
```
Comment (gziolo) [inline packages/scripts/utils/config.js]: "Skipping scanning" can probably be rephrased as it doesn't look right even for me while being non-native speaker 😅

Comment (sirreal) [reply packages/scripts/utils/config.js]: It's used in a couple of places. Maybe `Not scanning … due to …`.

**Review by sirreal — APPROVED** [review]
This looks good, thanks for these fixes 👍

File: packages/scripts/utils/config.js (hunk 1 above)
Comment (gziolo) [reply packages/scripts/utils/config.js]: Done with https://github.com/WordPress/gutenberg/pull/66752/commits/c86a198f938a97329cbb1c0a2f666202225abe4a.

**Pull request merged**